convert_kinbody_to_urdf.py

```diff
@@ -0,0 +1,194 @@
+# convert_kinbody_to_urdf.py
+
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+
+class Kinbody_Converter:
+    def __init__(self):
+        self.kinbody = None
+        self.urdf = None
+        self.link = None
+
+    def parse_from_file(self, filename):
+        if not filename.endswith('.kinbody.xml'):
+            print('Invalid input file: %s' % filename)
+            return False
+
+        self.kinbody = ET.parse(filename)
+        if self.kinbody is None:
+            print('Could not open file: %s' % filename)
+            return False
+
+        print('read kinbody from  %s' % filename)
+        return True
+
+    # TODO:
+    # handle Quat, RotationAxis
+    # handle sphere with Render and 0 radius
+    def convert_child(self, parent, level=0):
+        # print(level, parent.tag, parent.attrib, parent.text.strip())
+        if parent.tag.lower() == 'geom':
+            gtype = parent.attrib['type']
+            features = dict()
+            for child in parent:
+                if child.tag.lower() == 'extents':
+                    extents = map(float, child.text.strip().split(' '))
+                    extents_str = ''
+                    for val in extents:
+                        tval = '%.5f' % (float(val) * 2)
+                        extents_str += tval.rstrip('0') + ' '
+                    features[child.tag.lower()] = extents_str.strip()
+                else:
+                    features[child.tag.lower()] = child.text.strip()
+
+            collision = self.add_sub_element(self.link, 'collision')
+            visual = self.add_sub_element(self.link, 'visual')
+
+            if gtype == 'trimesh':
+                self.add_mesh_data(collision, features['data'])
+                self.add_mesh_render(visual, features['render'])
+
+            else:
+                self.add_origin(collision, features['translation'])
+                self.add_origin(visual, features['translation'])
+                self.add_color(visual, features)
+
+                if gtype == 'box':
+                    self.add_box(collision, features['extents'])
+                    self.add_box(visual, features['extents'])
+                elif gtype == 'cylinder':
+                    self.add_cylinder(collision,
+                                      features['radius'],
+                                      features['height'])
+                    self.add_cylinder(visual,
+                                      features['radius'],
+                                      features['height'])
+                elif gtype == 'sphere':
+                    self.add_sphere(collision, features['radius'])
+                    self.add_sphere(visual, features['radius'])
```

Comment:
Yes, you are right. So far I tried to cover the minimum set of elements and behaviors that are used in kinbody files in objects directory. If this converter would be useful, I will add the full sub-elements like quat, rotationaxis, rotationmat, vertices and special rendering cases.